prov/efa/src/rdm/efa_rdm_msg.c

```diff
@@ -557,10 +553,6 @@ ssize_t efa_rdm_msg_tinject(struct fid_ep *ep_fid, const void *buf, size_t len,
 	struct efa_rdm_peer *peer;
 
 	efa_rdm_ep = container_of(ep_fid, struct efa_rdm_ep, base_ep.util_ep.ep_fid.fid);
-	if (len > efa_rdm_ep->inject_size) {
-		EFA_WARN(FI_LOG_CQ, "invalid message size %ld for inject.\n", len);
-		return -FI_EINVAL;
-	}
 
```

Comment:
you need an assert here right?

Reply:
I omitted these since the documentation for `len` in fi_tagged(3) doesn't have the same qualifier re: valid lengths, as in fi_msg(3) and fi_rma(3). This could easily be an oversight since the documentation was writted ~10 years ago and never changed. WDYT?

Reply:
I think it's clearly an oversight in the documentation (cc @j-xiong ). You can add assert and update the man page

Reply:
Agree.